### ISSUE TITLE
maven plugin: unable to configure scanset from the command line

### ISSUE THREAD

**Issue by aw-was-here** [dependency-check/DependencyCheck#1425]
I might be doing something wrong, but it doesn't appear setting the scanSet from the command line of the maven plugin works:

```
mvn --batch-mode -DscanSet=src --DversionCheckEnabled=false -DskipProvidedScope=true -DskipSystemScope=true org.owasp:dependency-check-maven:3.3.0:check 
....
[ERROR] Failed to execute goal org.owasp:dependency-check-maven:3.3.0:check (default-cli) on project hadoop-main: Unable to parse configuration of mojo org.owasp:dependency-check-maven:3.3.0:check for parameter #: Cannot find default setter in class org.apache.maven.shared.model.fileset.FileSet -> [Help 1]
...
```

My ultimate goal is to scan "every directory but target", but I can't seem to get a basic 'src' working. 
Thanks!

**Comment by jeremylong** [comment]
As the ScanSet is a complex data type - I have no idea how one would set that via the command line. The configuration can be added to the `pom.xml` as:

```xml
<configuration>
   <scanSet>
      <fileSet>
         <directory>src</directory>
      </fileSet>
   </scanSet>
</configuration>
```

**Comment by aw-was-here** [comment]
I'm trying to do scans without modifying the project as part of adding dependency checker support to some Apache Software Foundation build and release tools (specifically under the Apache Yetus umbrella).  Adding support to Yetus will automate scans for a variety of ASF projects (Apache Hadoop, Apache HBase, and more).

The non-plugin doesn't do maven dependencies and the maven plugin doesn't scan every potentially relevant file without modifying the pom.  Currently, to get a full list of potential vulnerabilities, it means running dependency checker in its "two modes" (i.e., effectively twice).  If the maven plugin had the ability to specify scan sets from the CLI (and, even better, an "everything but" scan set), this would eliminate the need for two runs.

**Comment by wlfshmn** [comment]
I too have this issue where I invoke the maven plugin as part of a jenkins pipeline. The project is well served by the plugin for those dependencies expressed through the pom, but other parts of the build aggregates OSGi bundles through karaf feature files, which are not detected by the plugin (and can't reasonably be expected to be). The ability to point to various locations or results from the command-line would be beneficial to me too.

**Comment by jeremylong** [comment]
The functionality to specify other scan sets exists. If there are organizational issues with modifying the pom.xml (or even the parent pom.xml) to add additional configuration for dependency-check is a different problem.  Have you considered putting this configuration into a profile so most developers do not run dependency-check as part of the build?

Failing that - PRs are welcome. This would be a fairly low priority item for the dependency-check team as there are a ton of other enhancement requests.

**Comment by DominicDeCesare-Capita** [comment]
-DscanSet.fileSet=['src/main'] worked for me

**Comment by jeremylong** [comment]
The previous code to resolve this issue is not working. Please see #2671 for the new configuration option that will be available when 6.0 is released.
